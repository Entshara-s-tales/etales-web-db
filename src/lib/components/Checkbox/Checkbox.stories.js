import Checkbox from './Checkbox.svelte';

export default {
  title: 'lib/components/Checkbox',
  component: Checkbox,
  argTypes: {},
};

const Template = ({ onClick, ...args }) => ({
  Component: Checkbox,
  props: args,
});

export const SampleUsage = Template.bind({});
SampleUsage.args = {
  label: 'Checkbox',
};
