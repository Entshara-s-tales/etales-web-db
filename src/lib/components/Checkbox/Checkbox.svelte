<script lang="ts">
	export let name;
	export let label;
</script>

<label class="flex justify-start items-start" for={name}>
	<div
		class="bg-white border-2 rounded border-gray-400 w-6 h-6 flex flex-shrink-0 justify-center items-center mr-2 focus-within:border-blue-500"
	>
		<input {name} id={name} type="checkbox" class="opacity-0 absolute" />
		<svg class="fill-current hidden w-4 h-4 text-green-500 pointer-events-none" viewBox="0 0 20 20"
			><path d="M0 11l2-2 5 5L18 3l2 2L7 18z" /></svg
		>
	</div>
	<div class="select-none">{label}</div>
</label>

<style>
	input:checked + svg {
		display: block;
	}
</style>
