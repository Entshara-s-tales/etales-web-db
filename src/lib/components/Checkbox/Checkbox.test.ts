import { render } from '@testing-library/svelte';
import Checkbox from './Checkbox.svelte';

describe('<Checkbox/>', () => {
	it('matches snapshot', () => {
		const results = render(Checkbox, {
			name: 'checkbox-name',
			label: 'MyLabel',
		});
		expect(results.container).toMatchSnapshot();
	});
});
