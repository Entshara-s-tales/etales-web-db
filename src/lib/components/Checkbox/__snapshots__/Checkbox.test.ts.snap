// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox/> matches snapshot 1`] = `
<body>
  <div>
    <label
      class="flex justify-start items-start"
      for="checkbox-name"
    >
      <div
        class="bg-white border-2 rounded border-gray-400 w-6 h-6 flex flex-shrink-0 justify-center items-center mr-2 focus-within:border-blue-500"
      >
        <input
          class="opacity-0 absolute svelte-hy0kbl"
          id="checkbox-name"
          name="checkbox-name"
          type="checkbox"
        />
         
        <svg
          class="fill-current hidden w-4 h-4 text-green-500 pointer-events-none svelte-hy0kbl"
          viewBox="0 0 20 20"
        >
          <path
            d="M0 11l2-2 5 5L18 3l2 2L7 18z"
          />
        </svg>
      </div>
       
      <div
        class="select-none"
      >
        MyLabel
      </div>
    </label>
  </div>
</body>
`;
